src/server/procedure/procedure-with-graphql-client.ts

```diff
@@ -0,0 +1,22 @@
+import { MiddlewareFunction, TRPCError } from "@trpc/server";
+import { middleware, procedure } from "../server";
+import { saleorApp } from "../../../saleor-app";
+import { createClient } from "../../lib/graphql";
+import { attachAppToken } from "../middleware/attach-app-token";
```

Comment:
I see there are a lot of relative paths in the project, maybe we can introduce alias `@/...` or `@saleor/...` as baseUrl or in paths in tsconfig?

Reply:
I know this is a convention in Dashboard, but I don't want to introduce it in app-template:
1. App template is small and apps are small in general. Abstraction has a cost and I don't think its paid off here
2. App template philosophy is to be lean and unopinionated. It should be *a regular next.js project* + some preconfigured stuff Saleor *needs*. 
3. In examples we keep only code required by the example itself, so developers should see the diff and know "this code I need". If we introduce more, it will be confusing

I believe everyone can configure project after bootstrap just like s/he likes, but Saleor shouldn't enforce it. That's why we don't enforce linters etc too.

Reply:
Also, I don't care about "../../" because IDE (and even TS) is providing autocomplete

Reply:
Pros in skipping the `../../` part I see are:
- possibility to see in the imports clearly files structure from which things are imported, especially when long backward lookups are involved like `../../../../`
- refactor like moving files from one place to another do not require rewriting imports (although current IDEs looks like they do it pretty well for you, so this cons might not even appear)
- I'm not sure if it is legitimate, but maybe it may help with resolving circular imports issues, which might be a nightmare to fix because of probable lack of realization that it exists

But I understand that it is only template/example, so yeah, leaving it seems also reasonable for me.